examples/dcgan/dcgan.gop

```diff
@@ -28,13 +30,108 @@ var (
 	cuda       = flag.Bool("cuda", false, "enables cuda")
 	dryRun     = flag.Bool("dry-run", false, "check a single training cycle works")
 	ngpu       = flag.Int("ngpu", 1, "number of GPUs to use")
-	netG       = flag.String("netG", "", "path to netG (to continue training)")
-	netD       = flag.String("netD", "", "path to netD (to continue training)")
+	pathG       = flag.String("pathG", "", "path to netG (to continue training)")
+	pathD       = flag.String("pathD", "", "path to netD (to continue training)")
 	outf       = flag.String("outf", ".", "folder to output images and model checkpoints")
 	manualSeed = flag.Int("manualSeed", 0, "manual seed")
 	classes    = flag.String("classes", "bedroom", "comma separated list of classes for the lsun data set")
 )
 
+
+type Generator struct {
+	ngpu int
+	main *torch.Module
```

Comment:
Why name the field main? If the purpose is to inherit methods from torch.Module, we don't need a field name here.

Reply:
It's a data member of `Generator` network, just following the Python codes.

```python
class Generator(nn.Module):
    def __init__(self, ngpu):
        super(Generator, self).__init__()
        self.ngpu = ngpu
        self.main = nn.Sequential(
            # input is Z, going into a convolution
            nn.ConvTranspose2d(     nz, ngf * 8, 4, 1, 0, bias=False),
            nn.BatchNorm2d(ngf * 8),
            nn.ReLU(True),
            # state size. (ngf*8) x 4 x 4
            nn.ConvTranspose2d(ngf * 8, ngf * 4, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ngf * 4),
            nn.ReLU(True),
            # state size. (ngf*4) x 8 x 8
            nn.ConvTranspose2d(ngf * 4, ngf * 2, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ngf * 2),
            nn.ReLU(True),
            # state size. (ngf*2) x 16 x 16
            nn.ConvTranspose2d(ngf * 2,     ngf, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ngf),
            nn.ReLU(True),
            # state size. (ngf) x 32 x 32
            nn.ConvTranspose2d(    ngf,      nc, 4, 2, 1, bias=False),
            nn.Tanh()
            # state size. (nc) x 64 x 64
        )

    def forward(self, input):
        if input.is_cuda and self.ngpu > 1:
            output = nn.parallel.data_parallel(self.main, input, range(self.ngpu))
        else:
            output = self.main(input)
        return output
```